<a href="https://colab.research.google.com/github/Lordetimy/test2/blob/main/Kursovaya.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#@title Import
import math
from scipy.optimize import fsolve

In [ ]:
#@title Переменные
#D-Годовой спрос
#H-затраты хранения на ед.
#S-Издержки на размещение заказа
#Dd -затраты хранения на ед.(в день)
#Sd - стандартное отклонение спроса (день)
#Psl -желаемый уровень обслуживания
#L - время выполнения заказа
#T - срок между заказами 
#m - кол-во дней в году
#SS - страховой запас
#EOQ - экономический размер заказа
#Qreal - объем заказа
#N - число заказов в год
#TH -  издержки хранения
#TS - издержки заказа
#TSH = TS+TH (полные издержки)
#Sl - Стандартное откл. спроса за время выполнения заказа
#Esl - Кол-во необслуженных клиентов
#z - отклонение запаса от среднего, обеспечивающий заданный риск дефицита
#a - риск дефицита
#ROP - точка перезаказа
#TL = T+L  (полное время для получения заказа)
#I - остатки на складе
#Slt - Стандартное откл спроса за время выполнения заказа и время между заказами
#Qftmp - необходимый заказ

In [ ]:
#@title Переменные из уcловий
D=2400
H=4
S=5
Sd=4
Psl=0.98
L=7
m=365
Qreal=100



In [ ]:
#@title вычисленные переменные 1
N=D/Qreal
Dd=D/m
T=m/N
EQQ=math.sqrt(D*S*2/H)
TH=H*Qreal/2
TS=D*S/Qreal
TSH=TH+TS
Sl=Sd*math.sqrt(L)
Esl=(1-Psl)*Qreal


In [ ]:
#@title Решение 1
def equations(vars):
    a, z = vars
    eq1 = Sl*(1/math.sqrt(2*math.pi)*math.exp(-z**2/2)-z*a)-Esl
    eq2 = a-(1-(1+math.erf(z/math.sqrt(2)))/2)
    return [eq1, eq2]

a, z =  fsolve(equations, (0,0))
ROP=Dd*L+Sl*z
print("Точка перезаказа: ", ROP)

Точка перезаказа:  51.6263054272888


In [ ]:
#@title вычесленные переменные 2
Qreal2=77
N2=D/Qreal2
T2=m/N2
TH2=H*Qreal2/2
TS2=D*S/Qreal2
TSH2=TH2+TS2
Esl2=(1-Psl)*Qreal2
DATA=[N2,T2, TH2, TS2, TSH2, Esl2]


In [ ]:
#@title Решение 2
def equations(vars):
    a, z = vars
    eq1 = Sl*(1/math.sqrt(2*math.pi)*math.exp(-z**2/2)-z*a)-Esl2
    eq2 = a-(1-(1+math.erf(z/math.sqrt(2)))/2)
    return [eq1, eq2]

a2, z2 =  fsolve(equations, (0,0))
ROP2=Dd*L+Sl*z2
print("Точка перезаказа 2: ", ROP2)

Точка перезаказа 2:  53.32096804010094


In [ ]:
#@title вычесленные переменные 3
TL=round(T+L)
I=round(ROP,1)
Slt=Sd*math.sqrt(TL)

In [ ]:
#@title Решение 3
def equations(vars):
    a, z = vars
    eq1 = Slt*(1/math.sqrt(2*math.pi)*math.exp(-z**2/2)-z*a)-Esl
    eq2 = a-(1-(1+math.erf(z/math.sqrt(2)))/2)
    return [eq1, eq2]

a3, z3 =  fsolve(equations, (0,0))
Qftpm=Dd*TL+z3*Slt-I
print("Необходимый заказ: ", Qftpm)

Необходимый заказ:  109.32886312231119


In [ ]:
#@title Решение 4
Qftmp2=100
def equations(vars):
    a, z, Psl, Esl= vars
    eq1 = Slt*(1/math.sqrt(2*math.pi)*math.exp(-z**2/2)-z*a)-(Esl)
    eq2 = a-(1-(1+math.erf(z/math.sqrt(2)))/2)
    eq3 = Psl-(1-Esl/Qftmp2)
    eq4= Qftmp2-Dd*TL-z*Slt+I
    return [eq1, eq2, eq3, eq4]
a4, z4, Psl4, Esl4 =  fsolve(equations, (0,0,0,0))
print("Сервийсный уровень = ",round(Psl4*100, 2),"%")

Сервийсный уровень =  95.48 %


Ответы на вопросы:


1.   точка перезаказа = 52
2.   экономический размер заказа =72, точка перезаказа = 54
3.   необходимый заказ = 110, увеличилось на 10 ед., связано с переходом от модели фиксированного размерв заказа к модели с фиксированным периудом между заказами.
4.   сервисный уровень = 95.48%

